src/tqec/simulation/io_utils/csv_files.py

```diff
@@ -0,0 +1,55 @@
+"""Read and write simulation data from and to csv files."""
+
+import io
+from collections.abc import Iterable
+from pathlib import Path
+from typing import Literal, TextIO
+
+import sinter
+
+
+def write_sinter_stats_to_csv(
+    filepath: str | Path,
+    stats: Iterable[sinter.TaskStats],
+    if_file_exists: Literal["merge"]
+    | Literal["overwrite"]
+    | Literal["raise"] = "raise",
+) -> None:
+    """Write simulation data to csv file.
+
+    This is a convenience function because `sinter` does not expose this functionality. To read
+    simulation data use `sinter.read_stats_from_csv_files`.
+
+    Args:
+        filepath: The file to be written to.
+        stats: List of simulation results. Each element corresponds to a line in a csv file.
+        if_file_exists: How to react if `filepath` points to an existing file.
+            - raise: Raise an `AssertionError` if the file exists.
+            - overwrite: Effectively delete the file and pretend none exists.
+            - merge: Read the file and try to match entries with the simulation data from `stats`
+                by strong id. In case of a match merge the entries.
+    """
+    filepath = Path(filepath)
+
+    if filepath.exists():
+        if if_file_exists == "raise":
+            raise AssertionError(f"File `{filepath}` already exists.")
+        elif if_file_exists == "overwrite":
+            # Overwrite happens later automatically.
+            pass
+        elif if_file_exists == "merge":
+            # Writing stats to a file is necessary because sinter does not expose merging via API.
+            # We can only indirectly merge by reading in two files.
+            in_memory_file = io.StringIO()
+            _write_sinter_stats(in_memory_file, stats)
+            _ = in_memory_file.seek(0)
+            stats = sinter.read_stats_from_csv_files(filepath, in_memory_file)
+
+    with open(filepath, "w") as f:
+        _write_sinter_stats(f, stats)
```

Comment:
I do not understand how that merges. For me here is what is done when `if_file_exists == "merge"` is `True`:

1. write the user-provided `stats` in a file and read them back (never touching the existing file at `filepath`), so basically the identity operation (modulus merging some of the provided `stats` between each other).
2. overwrite the existing stats in `filepath` with the provided `stats`.

If I understood correctly how `sinter` works, merging two stats is equivalent to writing both of them in the same file. So, if the previous is right, it should be sufficient to simply append the stats to the existing file.

```suggestion
        elif if_file_exists == "overwrite":
            filemode = "w"
        elif if_file_exists == "merge":
            filemode = "a"
            
    with open(filepath, filemode) as f:
        _write_sinter_stats(f, stats)
```

To be tested though, I am not 100% sure of that.

Reply:
I missed the two filepaths in `stats = sinter.read_stats_from_csv_files(filepath, in_memory_file)`. You can disregard my comment :)